vllm/sequence.py

```diff
@@ -62,6 +66,14 @@ def __init__(
         self.output_token_ids: List[int] = []
         self.cumulative_logprob = 0.0
 
+        self.prompt_logprobs: Optional[List[Optional[float]]] = None
+        self.prompt_top_logprobs: Optional[List[Optional[Dict[int,
+                                                              float]]]] = None
```

Comment:
Let's move these two properties under the `Sequence` class instead of the current `SequenceData` class? The `SequenceData` class only keeps the bare minimal set of data to be the input of the neural network. We need to keep this class small since this class may need to be passed through RPC.

Reply:
This is actually a workaround that I couldn't figure out a better solution without more intrusive modifications. The way I implemented getting logprobs is to gather these prompt logprobs the first time calling sampler. However, from the view of `Sampler`, the `forward()` method doesn't have access to `Sequence`, making it unable to pass the gathered prompt logprobs back to the `Sequence`. 

One improvement I can think of is to keep these fields in `SequenceData` and make them also available in `Sequence`. Once the `fork()` method finds that they are set in corresponding `SequenceData`, it can pull them out from the `SequenceData` to `Sequence` so that they won't be further passed via network. Please let me know if you got a better thought on that.

Reply:
Instead of directly modifying these fields in the `Sampler`, can we return the probabilities in the `SequenceOutputs` class and modify the `Sequence` class later?

Reply:
Ah, I suppose that I have already implemented that.... 🤣 the thing I described was in a previous implementation.... 

Ok, I'm gonna move it to `Sequence`. It should work directly.